### ISSUE TITLE
Crash on open vault

### ISSUE THREAD

**Issue by mamoit** [flathub/md.obsidian.Obsidian#3]
Running version `0.9.10` on Kubuntu 18.04.5
Crashes as soon as I press the "open" vault button on the main window or "browse" on the create new vault menu.
Appimage works well.

```
$ flatpak run md.obsidian.Obsidian
Could not get Obsidian config. Skipping trash behavior update.
(electron) 'allowRendererProcessReuse' is deprecated and will be removed.
Loading main app package /app/resources/obsidian.asar
Checking for update using Github
Success.
App is up to date.

(obsidian:4): Gdk-ERROR **: 12:03:25.489: The program 'obsidian' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadShmSeg (invalid shared segment parameter)'.
  (Details: serial 1623 error_code 128 request_code 130 (MIT-SHM) minor_code 3)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
[fake-sandbox: zypak-sandbox] No data could be read (host died?)
[fake-sandbox: zypak-sandbox] Quitting Zygote...
/app/bin/obsidian.sh: line 3:     4 Trace/breakpoint trap   zypak-wrapper /app/obsidian $@
```

Let me know if more details are needed.

**Comment by mamoit** [comment]
Added `share=ipc` on flatseal and it's now working.

**Comment by joshua-stone** [comment]
@mamoit Can you provide more detailed instructions on how to replicate this behavior, as well as the flatpak version? I'm trying to trigger this bug in Kubuntu 18.04.5 but Obsidian isn't crashing:

```
user@testvm:~$ flatpak --version
Flatpak 1.0.9
user@testvm:~$ flatpak run md.obsidian.Obsidian
(electron) 'allowRendererProcessReuse' is deprecated and will be removed.
Loading main app package /app/resources/obsidian.asar
Checking for update using Github
Success.
App is up to date.
```

**Comment by mamoit** [comment]
I'm using flatpak from the official ppa as described [here](https://www.flatpak.org/setup/Ubuntu/) (version 1.8.2).
> With older Ubuntu versions, the official Flatpak PPA is the recommended way to install Flatpak.

If you already have it configured to open a vault it won't crash on startup, press "open another vault" > "open".
If I disable `share=ipc` it crashes consistently, if I enable it on the other hand, everything works fine and the file browser pops up as expected.

**Comment by mamoit** [comment]
It seems, from what I gather, that [if you are running an X11 program, you really need IPC](https://docs.flatpak.org/en/latest/sandbox-permissions.html#standard-permissions).
> --share=ipc - share IPC namespace with the host (necessary for X11)

**Comment by joshua-stone** [comment]
@mamoit I've merged your fix to go with the update to 0.9.11. Please let me know if this fixes the issue!

**Comment by mamoit** [comment]
Working well :+1:
No more crash when the file browser opens.
Thanks, now I'll go and explore the shiny new features :smile:
